src/UI/Art/SpriteArtist/OverlayArtist.java

```diff
@@ -0,0 +1,110 @@
+package UI.Art.SpriteArtist;
+
+import java.awt.AlphaComposite;
+import java.awt.Color;
+import java.awt.Composite;
+import java.awt.Graphics;
+import java.awt.Graphics2D;
+import java.awt.image.BufferedImage;
+import java.util.ArrayList;
+import java.util.Collection;
+
+import Engine.XYCoord;
+import Terrain.GameMap;
+import UI.GameOverlay;
+
+public class OverlayArtist
+{
+  private static final Color HIGHLIGHT_COLOR = new Color(255, 255, 255, 160); // white
+
+  public static final Color MOBILE_FIRE_EDGE = new Color(240,   0,   0, 255); // red
+  public static final Color SIEGE_FIRE_EDGE  = new Color(  0,   0, 160, 255); // dark blue
+  public static final Color FIRE_FILL        = new Color(  0,   0,   0,   0);
+
+  private static final int OVERLAY_EDGE_THICKNESS = 2;
+
+  static AlphaComposite buffComposite = null;
+  static long lastCompositeCreationTime = 0;
+
+  public static void drawHighlights(Graphics bigG,
+                                    GameMap gameMap,
+                                    Collection<GameOverlay> inputOverlays,
+                                    int drawX, int drawY,
+                                    int viewWidth, int viewHeight,
+                                    int tileSize)
+  {
+    if( viewWidth < 1 || viewHeight < 1 )
+      return;
+
+    BufferedImage overlayImage = SpriteLibrary.createTransparentSprite(viewWidth, viewHeight);
+    Graphics og = overlayImage.getGraphics();
+
+    // Only show stuff we're allowed to
+    Collection<GameOverlay> overlays = new ArrayList<GameOverlay>();
+    for( GameOverlay ov : inputOverlays )
+      if( null == ov.origin || !gameMap.isLocationFogged(ov.origin) )
```

Comment:
This check would be more useful if the `GameOverlay` was ever constructed with `origin != null`.

Reply:
It sure *should* be.
...I'm an idiot. I didn't. I blame sleep, and the lack of it.